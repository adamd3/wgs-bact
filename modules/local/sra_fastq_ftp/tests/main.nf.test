nextflow_process {

    name "Test process: SRA_FASTQ_FTP"
    script "../main.nf"
    process "SRA_FASTQ_FTP"
    tag "modules"
    tag "modules_local"
    tag "sra_fastq_ftp"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }

            process {
                """
                input[0] = [
                    [ id:'SRX9626017_SRR13191702', single_end:false, md5_1: '89c5be920021a035084d8aeb74f32df7', md5_2: '56271be38a80db78ef3bdfc5d9909b98' ], // meta map
                    params.test_data['modules_local']['sra_fastq_ftp']
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
