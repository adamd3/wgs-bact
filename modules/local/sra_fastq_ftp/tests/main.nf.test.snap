{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,89c5be920021a035084d8aeb74f32df7",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,56271be38a80db78ef3bdfc5d9909b98"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,055a6916ec9ee478e453d50651f87997",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,c30ac785f8d80ec563fabf604d8bf945"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,6b9d69dea1c1305f74a65197ee871f1b"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,89c5be920021a035084d8aeb74f32df7",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,56271be38a80db78ef3bdfc5d9909b98"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "single_end": false,
                            "md5_1": "89c5be920021a035084d8aeb74f32df7",
                            "md5_2": "56271be38a80db78ef3bdfc5d9909b98"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,055a6916ec9ee478e453d50651f87997",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,c30ac785f8d80ec563fabf604d8bf945"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6b9d69dea1c1305f74a65197ee871f1b"
                ]
            }
        ],
        "timestamp": "2023-09-22T10:58:46.998421"
    }
}